runtime-modules/content/src/nft/types.rs

```diff
@@ -169,391 +83,338 @@ impl<MemberId> Default for NftOwner<MemberId> {
-    pub fn ensure_caller_is_last_bidder<T: Trait>(&self, who: MemberId) -> DispatchResult {
-        ensure!(self.is_last_bidder(who), Error::<T>::ActorIsNotALastBidder);
-        Ok(())
+        Self {
+            end,
+            auction_duration,
+            top_bid: Some(EnglishBid { amount, bidder_id }),
+            ..self
+        }
     }
+}
 
-    /// Ensure auction type is `Open`
-    pub fn ensure_is_open_auction<T: Trait>(&self) -> DispatchResult {
-        ensure!(
-            matches!(&self.auction_type, AuctionType::Open(_)),
-            Error::<T>::IsNotOpenAuctionType
-        );
-        Ok(())
+/// Open Auction
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct OpenAuctionRecord<BlockNumber, AuctionId, Balance, MemberId: Ord> {
+    pub starting_price: Balance,
+    pub buy_now_price: Option<Balance>,
+    pub whitelist: BTreeSet<MemberId>,
+    pub bid_lock_duration: BlockNumber,
+    pub auction_id: AuctionId,
+}
+
+impl<
+        BlockNumber: Copy + Saturating + PartialOrd,
+        AuctionId: Copy,
+        Balance: Copy + PartialOrd + Saturating,
+        MemberId: Ord + Copy,
+    > OpenAuctionRecord<BlockNumber, AuctionId, Balance, MemberId>
+{
+    pub fn new(
+        params: OpenAuctionParamsRecord<BlockNumber, Balance, MemberId>,
+        auction_nonce: AuctionId,
+    ) -> Self {
+        Self {
+            starting_price: params.starting_price,
+            buy_now_price: params.buy_now_price,
+            whitelist: params.whitelist.clone(),
+            bid_lock_duration: params.bid_lock_duration,
+            auction_id: auction_nonce,
+        }
     }
 
-    /// Ensure auction type is `English`
-    pub fn ensure_is_english_auction<T: Trait>(&self) -> DispatchResult {
+    pub(crate) fn ensure_whitelisted_participant<T: Trait>(
+        &self,
+        participant_id: MemberId,
+    ) -> DispatchResult {
         ensure!(
-            matches!(&self.auction_type, AuctionType::English(_)),
-            Error::<T>::IsNotEnglishAuctionType
+            self.whitelist.is_empty() || self.whitelist.contains(&participant_id),
+            Error::<T>::MemberIsNotAllowedToParticipate
         );
         Ok(())
     }
 
-    /// Ensure bid lock duration expired
-    pub fn ensure_bid_lock_duration_expired<T: Trait>(
+    pub(crate) fn ensure_auction_has_no_bids<T: Trait>(
         &self,
-        current_block: BlockNumber,
-        bid: Bid<MemberId, AccountId, BlockNumber, Balance>,
+        video_id: T::VideoId,
     ) -> DispatchResult {
-        if let AuctionType::Open(OpenAuctionDetails { bid_lock_duration }) = &self.auction_type {
-            ensure!(
-                current_block - bid.made_at_block >= *bid_lock_duration,
-                Error::<T>::BidLockDurationIsNotExpired
-            );
-        }
+        ensure!(
+            crate::BidByVideoAndMember::<T>::iter_prefix(video_id)
```

Comment:
This is not workable, not safe, not needed.

Reply:
Removed here e256eba1394bd5ce6fe5bde99645c90cde5b36e8